### ISSUE TITLE
[a11y]: Focus is moving out of tearsheet

### ISSUE THREAD

**Issue by Rakesh-Addepalle** [carbon-design-system/ibm-products#6164]
### Package

@carbon/ibm-products

### Browser

Chrome, Firefox

### Operating System

MacOS

### Package version

2.50.0

### React version

18.3.1

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

[2.4.3: Focus Order](https://www.w3.org/WAI/WCAG21/Understanding/focus-order)

### WCAG 2.1 Violation

[2.4.3: Focus Order](https://www.w3.org/WAI/WCAG21/Understanding/focus-order)

### Reproduction/example

https://stackblitz.com/edit/github-cqjvhd?file=src%2FExample%2Ftable.jsx,src%2FExample%2FExample.jsx,src%2FExample%2F_example.scss

### Steps to reproduce

In order to replicate please follow the below steps.
1) Click on any row --> tearsheet opens
2) Use tab and navigate to continue button and click on it.
3) You can observe focus is moving out of the tearsheet.

In our current implementation we are depending on the response from api call, based on the response content is displayed. to mimic that used set timeout function and loading indicator.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by oliviaflory** [comment]
@Rakesh-Addepalle after clicking `Continue` in the tearsheet I would expect the tearsheet to close and the focus to return to the data table. Is that your expectation as well? 

It looks like when I click the `Continue` button the focus returns to the data table, but the tearsheet remains with only the cancel button as clickable.

https://github.com/user-attachments/assets/841367b6-cb80-4b07-9a82-3eec4846f448

**Comment by makafsal** [comment]
@Rakesh-Addepalle - I've fixed the issue of the focus moving out and raised the PR. 

Refer: [CreateTearsheet](https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-patterns-create-flows-createtearsheet--multi-step-tearsheet&globals=viewport:basic)
